### PR TITLE
Add build option for Android

### PR DESCRIPTION
**Pull request by eclipse95**
Hi,
This PR allows users to build binary for Android with a specific API.

Changelog :
* Modify ``Makefile`` for build with specific Android API/arch.
* Modify ``ndk-gcc``.
* Update ``Readme``.

### CODE REVIEW

File: ndk-gcc [hunk 1]
```diff
@@ -1,10 +1,11 @@
-#!/bin/sh
+#!/usr/bin/env bash
```
Comment (trufae) [inline ndk-gcc]: don't add bashisms

File: ndk-gcc [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
 # android gcc wrapper
 
 if [ -z "${NDK}" ]; then
-	NDK=~/Downloads/android-ndk-r10e
+	NDK=/root/android-sdk/ndk-bundle
```
Comment (trufae) [inline ndk-gcc]: this path asumes you are running the build as root, which is not desirable

Comment (trufae) [reply ndk-gcc]: you can just set the NDK env to override, keep the old value here

File: Makefile [hunk 3]
```diff
@@ -123,20 +123,26 @@ uninstall:
...
 
-NDK_ARCH=arm
-ANDROID_ARCHS=arm mips x86
-ANDROID_VERSION=kitkat
+NDK_ARCH=
+ANDROID_ARCHS=arm mips x86 x86_64 aarch64
+ANDROID_API=
+
+ifneq ($(NDK_ARCH), "")
+ ANDROID_ARCHS=$(NDK_ARCH)
```
Comment (trufae) [inline Makefile]: do not indent this line

File: Makefile [hunk 4]
```diff
@@ -123,20 +123,26 @@ uninstall:
...
 LOLLIPOP_CFLAGS=-DHAVE_FANOTIFY=1 -DHAVE_SYS_FANOTIFY=0
 
-NDK_ARCH=arm
-ANDROID_ARCHS=arm mips x86
-ANDROID_VERSION=kitkat
+NDK_ARCH=
+ANDROID_ARCHS=arm mips x86 x86_64 aarch64
+ANDROID_API=
+
+ifneq ($(NDK_ARCH), "")
```
Comment (trufae) [inline Makefile]: ifneq($(NDK_ARCH),)

the "" is a gnuism afaik and its not necessary

File: Makefile [hunk 5]
```diff
@@ -123,20 +123,26 @@ uninstall:
...
+		$(MAKE) andcompile ANDROID_API=$(ANDROID_API) NDK_ARCH=$$a ; \
 	done
 
-ll lollipop:
-	./ndk-gcc 21 -fPIC -pie $(LOLLIPOP_CFLAGS) $(CFLAGS) $(LDFLAGS) -o fsmon-and-$(NDK_ARCH) $(SOURCES)
-
-kk kitkat:
-	./ndk-gcc 19 -fPIC -pie $(KITKAT_CFLAGS) $(CFLAGS) $(LDFLAGS) -o fsmon-kitkat-$(NDK_ARCH) $(SOURCES)
+andcompile:
+	if [ $(ANDROID_API) -gt 21 ]; then \
```
Comment (trufae) [inline Makefile]: you can do this check at make level with `APIGT21=$(shell test $(ANDROID_API) -gt 21 && echo 1)` and avoid the backslashes

**Comment by eclipse95** [comment]
Hi @trufae,
The commit bbba9ac answers your review.

**Comment by trufae** [comment]
Thanks!

**Pull request merged**